Model/Customer.php

```diff
@@ -117,10 +120,7 @@ public function create(array $params)
     public function addCard($cardToken)
     {
         if (! $this->getId()) {
-            $this->create([
-                'email'       => $this->magentoCustomer->getEmail(),
-                'description' => trim($this->magentoCustomer->getFirstname() . ' ' . $this->magentoCustomer->getLastname())
-            ]);
```

Comment:
Easier if these params are inside the function rather then being passed in function parameters.
Moved it to the function.

Reply:
I think its interface is a bit strange where the method name says `"addCard"` then check if not `"this->getId()"` then `"this->create()"`.
```php
public function addCard($cardToken)
{
    if (! $this->getId()) {
        $this->create();
    }

   ...
}
```

Reply:
it is your code... @guzzilar

Reply:
I was also feeling off for this part too. But still, can't figure it out where is the proper place to have this one (unless we imitate OmisePHP library by having nested-classes – something like, `$this->customer->card->add()`)